pymeasure/instruments/nf/li5660.py

```diff
@@ -0,0 +1,445 @@
+        """ A property that control the input connector """,
+        validator=strict_discrete_set,
+        values={"A": "A",
+                "AB": "AB",
+                "C": "C",
+                "Current": "I",
+                "HF": "HF"},
+        map_values=True
+
+    )
+
+    reference_signal_source = Instrument.control(
+        ":ROUT2?", "ROUT2 %s",
+        """ A property that control the reference signal source """,
+        validator=strict_discrete_set,
+        values={"Ref In": "RINP",
+                "Internal": "IOSC",
+                "Signal": "SINP"},
+        map_values=True
+    )
+
+    reference_frequency_source = Instrument.control(
+        ":ROSC:SOUR?",":ROSC:SOUR %s",
+        """ A property that control the reference frequency source for frequency synthesis.""",
+        validator=strict_discrete_set,
+        values={"Internal": "INT", "External": "EXT"},
+        map_values=True
+    )
+
+    ###############
+    # Sensitivity #
+    ###############
+    auto = Instrument.measurement(
+        ":AUTO:ONCE",
+        """ a property that automatically set the sensitivity and time constant once to match the
+        reference signal for the signal being measured at the time. Performs auto setting one time """
+    )
+
+    auto_current = Instrument.control(
+        "CURR:AC:RANG:AUTO?", "CURR:AC:RANG %d",
+        """ A property that continuous automatic selection current sensitivity function.
+        On(enabled), Off(disabled) and Once(auto sets current sensitivity one time).""",
+        values={"On": 1 , "Off": 0, "ONCE": "ONCE"},
+        map_values=True
+    )
+
+    currnet_range = Instrument.control(
+        ":CURR:AC:RANG?", ":CUR:AC:RANG:AUTO 0; :CURR:AC:RANG %g",
+        """ A property that control the current sensitivity (primary PSD) """,
+        validator=truncated_range,
+        values=[10e-15, 1e-6]
+    )
+
+    dynamic_reserve = Instrument.control(
+        "DRES?", "DRES %s",
+        """ A property that control the dynamic reserver sensitivity.
+            High dynamic reserve when noise level is high
+            Low dynamic reserve when noise level is low
+        """,
+        validator=strict_discrete_set,
+        values={"High": "HIGH", "Medium": "MED", "Low": "LOW"},
+        map_values=True
+    )
+
+    filter_slope = Instrument.control(
+        ":FILT:SLOP?", ":FILT:SLOP %d",
+        """ A property that control the filter attenuation slope (primary PSD). """,
+        validator=strict_discrete_set,
+        values=[6, 12, 18,24],
+        map_values=False
+    )
+
+    time_constant = Instrument.control(
+        ":FILT:TCON?",":FILT:TCON %g",
+        """ A property that control the filter time constant (primary PSD). """,
+        validator=truncated_range,
+        values=[1e-6, 50e+3]
+    )
+
+    filter_type = Instrument.control(
+        ":FILT:TYPE?",":FILT:TYPE %s",
+        """ A property that control the filter type (primary PSD). """,
```

Comment:
It would be nice to mention the possible values in the docstring.

Reply:
Filter type added